common/src/main/java/com/microsoft/identity/common/internal/platform/AsymmetricKey.java

```diff
@@ -0,0 +1,88 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.platform;
+
+import com.microsoft.identity.common.exception.ClientException;
+
+import java.util.Date;
+
+/**
+ * Represents an RSA asymmetric key. This object represents a single keypair instance inside of an
```

Comment:
If this *must be* an RSA key, I'd expect it to be in the name.  But I think that this interface can support any asymmetric key algorithm.  If so, we should weaken the documentation appropriately.

Reply:
Sure, will incorporate into the name. Some of this won't be visible once its pulled thru OneAuth (they will wrap in their own class), but that's not terribly important for our naming quality

Reply:
e76345c5d7f2884fa2eae6884186ef30606d13e7, 32e3fc7df261f5b9660f42e683b476e0bb38dff5